components/shared/base/id.rs

```diff
@@ -411,6 +412,19 @@ impl BlobId {
     }
 }
 
+namespace_id! {DomPointId, DomPointIndex, "DomPoint"}
+
+impl DomPointId {
+    pub fn new() -> DomPointId {
+        PIPELINE_NAMESPACE.with(|tls| {
+            let mut namespace = tls.get().expect("No namespace set for this thread!");
+            let next_point_id = namespace.next_dom_point_id();
+            tls.set(Some(namespace));
+            next_point_id
+        })
+    }
+}
+
```

Comment:
Will every "primitive" serializable type in the DOM need an ID type? It seems like we may end up with a lot of these as time goes on. Would it be possible for all types to share a single ID namespace?

Reply:
I have a branch that does this which I was going to provide separately, using this PR as further justification :)

Reply:
Sounds great!